src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -392,4 +440,26 @@ private boolean checkIfEqual(final Object element, final int index) {
                 || this.data[index] != null && data[index].equals(element);
 
     }
+
+    private void checkCollectionNotNull(final Collection<?> collection) {
+        if (collection == null) {
+            throw new NullPointerException("collection is null");
+        }
+    }
+
+    private boolean removeCollection(final Collection<?> collection,
+            final boolean remove) {
+        this.checkCollectionNotNull(collection);
+        boolean changed = false;
+
+        for (int i = 0; i < currentIndex; i++) {
+            if (collection.contains(this.data[i]) == remove) {
```

Comment:
What is complexity of contain here ? 

Can we guarantee O(1) here in contain method ??